format/format.go

```diff
@@ -235,12 +235,52 @@ func (p *Printer) applySpecialIndentRules(node *parse.ListNode) {
 			p.applySpecialLet(node.Nodes)
 		case IndentLetfn:
 			p.applySpecialLetfn(node.Nodes)
+		case IndentFor:
+			p.applySpecialFor(node.Nodes)
 		case IndentDeftype:
 			p.applySpecialDeftype(node.Nodes)
 		}
 	}
 }
 
+func (p *Printer) applySpecialForLet(nodes []parse.Node) {
+	i := -1
+	seenLet := false
+	for _, node := range nodes {
+		if !goclj.Semantic(node) {
+			continue
+		}
+		if seenLet {
+			if v, ok := node.(*parse.VectorNode); ok {
+				p.specialIndent[v] = indentBindings
+			}
+		}
+		seenLet = false
+		i++
+		if i%2 != 0 {
+			continue
+		}
+		if k, ok := node.(*parse.KeywordNode); ok {
+			if k.Val == ":let" {
+				seenLet = true
+			}
+		}
+	}
+}
+
+func (p *Printer) applySpecialFor(nodes []parse.Node) {
+	for _, node := range nodes[1:] {
+		if goclj.Newline(node) {
```

Comment:
Wouldn't you want to skip any non-semantic nodes here?

```
(for ; blah
  [x
     [1 2 3]]
  ...)
```

(Seems like some of the existing special forms have this mistake as well.)

Reply:
Makes sense, I replaced the rest of the places to skip non-semantic nodes instead of just newline nodes.